File: v6/prolog/compile/test/typegen_golden/shape_option_of_list.types.ts
export type Option<T> = { tag: 'none' } | { tag: 'some'; value: T };

export interface Record {
  id: number;
  tag_values: Option<Array<string>>;
  grid_values: Option<Array<Array<string>>>;
  note: Option<string>;
  maybe_tag_values: Array<Option<string>>;
}

File: v6/prolog/compile/test/typegen_golden/shape_option_of_list.types.rs
#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
#[serde(tag = "tag", content = "value", rename_all = "snake_case")]
pub enum DlOption<T> {
    None,
    Some(T),
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Record {
    pub id: i64,
    pub tag_values: DlOption<Vec<String>>,
    pub grid_values: DlOption<Vec<Vec<String>>>,
    pub note: DlOption<String>,
    pub maybe_tag_values: Vec<DlOption<String>>,
}
